go/plugins/dotprompt/genkit_test.go

```diff
@@ -73,3 +73,36 @@ func TestExecute(t *testing.T) {
 		t.Errorf("fake model replied with %q, want %q", got, want)
 	}
 }
+
+func TestExecuteWithModelName(t *testing.T) {
+	ai.DefineModel("test", "test", nil, testGenerate)
+	p, err := New("TestExecute", "TestExecute", Config{ModelName: "test/test"})
```

Comment:
@dhiaayachi 

It might be a bit hard to understand, but `ai.DefineModel()` is a function with side effects that internally registers the defined Model in a global scope registry. When doing so, if there is a duplicate Model with the same `[provider/model_name]`, an error is triggered, and the Model cannot be registered.

If you want the simplest fix, you can modify the test code as follows to ensure that the Models do not conflict:

```go
ai.DefineModel("test", "test2", nil, testGenerate)
p, err := New("TestExecute", "TestExecute", Config{ModelName: "test/test2"})
```

Alternatively, you can use something like [TestMain](https://pkg.go.dev/testing#hdr-Main) to ensure that `ai.DefineModel("test", "test", nil, testGenerate)` is called only once within the test file.

Reply:
This is the reason why the CI test failed.

Reply:
Thank you @yukinagae, I decided to use subtests to avoid having to define the model as a global var.